sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -27,27 +27,35 @@ def csr_row_norms(X):
     """L2 norm of each row in CSR matrix X."""
     if X.dtype not in [np.float32, np.float64]:
         X = X.astype(np.float64)
-    return _csr_row_norms(X.data, X.shape, X.indices, X.indptr)
+    return np.asarray(_csr_row_norms(X.data, X.indices, X.indptr))
 
 
-def _csr_row_norms(cnp.ndarray[floating, ndim=1, mode="c"] X_data,
-                   shape,
-                   cnp.ndarray[integral, ndim=1, mode="c"] X_indices,
-                   cnp.ndarray[integral, ndim=1, mode="c"] X_indptr):
+def _csr_row_norms(
+    floating[::1] X_data,
+    integral[::1] X_indices,
+    integral[::1] X_indptr,
+):
     cdef:
-        unsigned long long n_samples = shape[0]
-        unsigned long long i
+        integral n_samples = X_indptr.shape[0] - 1
```

Comment:
This does not seem to be correct:

```python
x = csr_matrix((3, 4), dtype=np.int8)
x[1, 3] = 10
x[2, 1] = 5
x.indptr.shape[0] - 1  # give 3 while having 2 samples
```

Reply:
`x` has 3 samples here and not 2.

Reply:
Oh right. I bugged with nonzero samples.